bin/deseq2_qc.r

```diff
@@ -71,18 +74,29 @@ if (file.exists(opt$outdir) == FALSE) {
 }
 setwd(opt$outdir)
 
-samples.vec <- sort(colnames(count.table))
-groups      <- sub("_[^_]+$", "", samples.vec)
-if (length(unique(groups)) == 1 || length(unique(groups)) == length(samples.vec)) {
-    quit(save = "no", status = 0, runLast = FALSE)
+samples.vec <- colnames(count.table)
+name_components <- strsplit(samples.vec, "_")
+n_components <- length(name_components[[1]])
+decompose <- n_components!=1 && all(sapply(name_components, length)==n_components) 
+coldata <- data.frame(samples.vec, sample=samples.vec, row.names=1)
+if (decompose) {
+  groupings <- as.data.frame(lapply(1:n_components, function(i) sapply(name_components, "[[", i)))
```

Comment:
2 space indents!! 👀

Reply:
It's the [R way](http://adv-r.had.co.nz/Style.html) - I'll fix it!